src/common/logger/logger_settings.cpp

```diff
@@ -0,0 +1,66 @@
+#include "pch.h"
+#include "logger_settings.h"
+#include <fstream>
+#include <Windows.h>
+#include <winrt/Windows.Data.Json.h>
+
+using namespace winrt::Windows::Data::Json;
+
+LogSettings::LogSettings()
+{
+    this->logLevel = LogSettings::defaultLogLevel;
+}
+
+std::optional<JsonObject> from_file(std::wstring_view file_name)
+{
+    try
+    {
+        std::ifstream file(file_name.data(), std::ios::binary);
+        if (file.is_open())
+        {
+            using isbi = std::istreambuf_iterator<char>;
+            std::string obj_str{ isbi{ file }, isbi{} };
+            return JsonValue::Parse(winrt::to_hstring(obj_str)).GetObjectW();
+        }
+        return std::nullopt;
+    }
+    catch (...)
+    {
+        return std::nullopt;
+    }
+}
+
+void to_file(std::wstring_view file_name, const JsonObject& obj)
+{
+    std::wstring obj_str{ obj.Stringify().c_str() };
+    std::ofstream{ file_name.data(), std::ios::binary } << winrt::to_string(obj_str);
+}
+
+JsonObject to_json(LogSettings settings)
+{
+    JsonObject result;
+    result.SetNamedValue(LogSettings::logLevelOption, JsonValue::CreateStringValue(settings.logLevel));
+
+    return result;
+}
+
+LogSettings to_settings(JsonObject jobject)
+{
+    LogSettings result;
+    result.logLevel = jobject.GetNamedString(LogSettings::logLevelOption);
```

Comment:
I think that throws if a user removes the log level from json file manually. let's add a default value here (LogSettings::defaultLogLevel?)

Reply:
Almost missed it